### PR TITLE
Generate runbook from mixins into markdown file

### PR DESCRIPTION
**Pull request by metalmatze**
This command `mixtool runbook` extracts multi line string descriptions from mixins and generates a markdown file.

Example alert object with runbook string:
```diff
{
  alert: 'KubeNodeNotReady',
  expr: |||
    kube_node_status_condition{%(kubeStateMetricsSelector)s,condition="Ready",status="true"} == 0
  ||| % $._config,
  labels: {
    severity: 'warning',
  },
  annotations: {
    message: 'Overcommited CPU resource requests on Pods, cannot tolerate node failure.',
  },
  'for': '1h',
+ runbook: |||
+   This is an awesome runbook text.
+ |||,
},
```

@brancz @squat

### CODE REVIEW

File: pkg/mixer/runbook.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+func Runbook(w io.Writer, filename string, opts RunbookOptions) error {
+	vm := jsonnet.MakeVM()
+
+	vm.Importer(&jsonnet.FileImporter{
+		JPaths: opts.JPaths,
+	})
+
+	snippetTemplate := `
+(import '%s') {
```
Comment (brancz) [inline pkg/mixer/runbook.go]: let's extract the majority of this into an actual `.libsonnet` file and then just end up with:

```
(import '%s') +
(import 'mixtool-internal/runbook.libsonnet') // this is an in memory provider for the jsonnet VM
```

Comment (metalmatze) [reply pkg/mixer/runbook.go]: Good idea!
I'm not 100% sure on what exactly you have in mind. Create this folder on disk and then use something like go-bindata (I'd prefer https://github.com/gobuffalo/packr/) to compile these files into the binary?

Comment (brancz) [reply pkg/mixer/runbook.go]: yes, compiling that into the binary is what I had in mind

**Comment by metalmatze** [comment]
I've updated this PR to now use packr's virtual FS with libsonnet templates when compiling.
PTAL.

**Comment by simonpasquier** [comment]
From my experience, `vfsgen` is pretty neat: its scope is well defined and it leverages only the standard library. `packr` seems to be very similar. The main difference I see between the 2 is that `vfsgen` doesn't even require you to install a binary for generating the assets (eg no `go get ...`).

**Comment by metalmatze** [comment]
That vfsgen doesn't need an extra binary sounds really good. I'll take a look. Thanks for your comment @simonpasquier!

**Comment by metalmatze** [comment]
I will merge this as it is for now, as we can easily change to vfsgen in the future.

**Pull request merged**